File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int i=0,j=0,maxOne=0;
        while(j<nums.size()&& i<nums.size()){
            if(nums[j]==0) k--;
            
            while(k<0 && j<nums.size()){
                if(nums[i]==0) k++;
                i++;
            }
            maxOne=max(maxOne,j-i+1);
            j++;
        }
        return maxOne;
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int len = nums.length;
        int start = 0;
        int noOfFlipsRemaining = k;
        int i=0,j=0;
        int overallMax = 0;
        while(i<=j && j<len){
            if(nums[j]==1){
            overallMax = Math.max(j-i+1,overallMax);      
                j++;
            } else {
                if(noOfFlipsRemaining>0){
                    nums[j] = 2;
                    noOfFlipsRemaining--;
                    overallMax = Math.max(j-i+1,overallMax);
                    j++;
                }else {
                    while(nums[i]==1){
                        i++;
                    }
                    if(i<len && nums[i]==2){
                        nums[i]=0;
                        i++;
                        noOfFlipsRemaining++;
                    }
                    if(i<len && i==j && nums[i]==0){
                        i++;
                        j++;
                    }
                }
            }
        }
        return overallMax;
    }
}